test/unit/eventsDirectiveSpec.js

```diff
@@ -20,7 +21,74 @@ describe('Directive: leaflet', function() {
         $rootScope.$apply();
     }));
 
-    it('should broadcast events from the rootscope when triggered leaflet events',function(){
+    ['', '1'].forEach(function(mapId){
+      it( mapId + ' should broadcast events from the rootscope when triggered leaflet events',function(){
+          var element = mapId? angular.element('<leaflet id="' + mapId + '" events="events"></leaflet>') :angular.element('<leaflet events="events"></leaflet>');
+          element = $compile(element)($rootScope);
+          var scope = element.scope();
+          var check = {};
+          var mapEvents = [
+              'click',
+              //'dblclick',
+              'mousedown',
+              'mouseup',
+              'mouseover',
+              'mouseout',
+              'mousemove',
+              'contextmenu',
+              'focus',
+              'blur',
+              'preclick',
+              'load',
+              'unload',
+              'viewreset',
+              'movestart',
+              'move',
+              //'moveend',
+              'dragstart',
+              'drag',
+              'dragend',
+              'zoomstart',
+              'zoomend',
+              'zoomlevelschange',
+              'resize',
+              'autopanstart',
+              //'layeradd',
+              //'layerremove',
+              'baselayerchange',
+              'overlayadd',
+              'overlayremove',
+              'locationfound',
+              'locationerror',
+              'popupopen',
+              'popupclose'
+          ];
+
+          function setEventTrue(origEventName, eventName) {
+              leafletLogger.debug(eventName + ' called.');
+              check[origEventName] = true;
+          }
+
+          if(mapId)
+            mapId = mapId + '.';
+
+          leafletData.getMap().then(function(map) {
+              mapEvents.forEach(function(origEventName){
+                  var eventName = 'leafletDirectiveMap.' + mapId + origEventName;
+                  // leafletLogger.log(eventName); // Inspect
+                  scope.$on(eventName, function(){
```

Comment:
Notice setEventTrue is wrapped in a function now as the original spec was generating false positives.